File: solutions/Cdgame/solve/cpp/CdgameSolver.cpp
#include <fstream>
#include <topcoder>
#include "Cdgame.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
    try {
        std::ifstream ifs(argv[1]);
        vector<int> a; tc::read(ifs, a); tc::next(ifs);
        vector<int> b; tc::read(ifs, b);
        ifs.close();

        std::ofstream ofs(argv[2]);
        Cdgame solver;
        tc::show(ofs, solver.rescount(a, b));
        ofs.close();
    } catch (std::exception &e) {
        std::cerr << e.what() << std::endl;
    }
    return 0;
}

File: solutions/Cdgame/solve/ruby/CdgameSolver.rb
#! /usr/bin/env ruby

require_relative "Cdgame"

require "gettc/types"
include Gettc

def init()
    gettc_home = File.join(ENV["HOME"], ".gettc")
    if ENV.has_key?("GETTC_HOME")
        gettc_home = ENV["GETTC_HOME"]
    end 
    gettc_home = File.expand_path(gettc_home)
    $LOAD_PATH << File.join(gettc_home, "include", "ruby")
    require "topcoder"
    include TopCoder
end

def main()
    reader = Reader.new(IO.read(ARGV[0]))
    a = reader.next(TArray.new(TInt))
    reader.next()
    b = reader.next(TArray.new(TInt))

    result = Cdgame.new().rescount(a, b)
    IO.write(ARGV[1], Writer.new().next(result, TInt).to_s())
end
init()
main()
